packages/plugin-ext/src/main/browser/window-activity-tracker.ts

```diff
@@ -0,0 +1,95 @@
+// terms of the Eclipse Public License v. 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0.
+//
+// This Source Code may also be made available under the following Secondary
+// Licenses when the conditions for such availability set forth in the Eclipse
+// Public License v. 2.0 are satisfied: GNU General Public License, version 2
+// with the GNU Classpath Exception which is available at
+// https://www.gnu.org/software/classpath/license.html.
+//
+// SPDX-License-Identifier: EPL-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0
+// *****************************************************************************
+
+import { Disposable, Emitter, Event } from '@theia/core';
+
+const CHECK_INACTIVITY_LIMIT = 2;
+const CHECK_INACTIVITY_INTERVAL = 10000;
+
+const eventListenerOptions: AddEventListenerOptions = {
+    passive: true,
+    capture: true
+};
+export class WindowActivityTracker implements Disposable {
+
+    private inactivityTime = 0; // number of times inactivity was checked since last reset
+    private readonly inactivityLimit = CHECK_INACTIVITY_LIMIT; // number of inactivity checks done before sending inactive signal
+    private readonly checkInactivityInterval = CHECK_INACTIVITY_INTERVAL; // check interval in milliseconds
+    private interval: NodeJS.Timeout | undefined;
+
+    protected readonly onDidChangeActiveStateEmitter = new Emitter<boolean>();
+    private _activeState: boolean = true;
+
+    constructor(readonly win: Window) {
+        this.initializeListeners(this.win);
+    }
+
+    get onDidChangeActiveState(): Event<boolean> {
+        return this.onDidChangeActiveStateEmitter.event;
+    }
+
+    private set activeState(newState: boolean) {
+        if (this._activeState !== newState) {
+            this._activeState = newState;
+            this.onDidChangeActiveStateEmitter.fire(this._activeState);
+        }
+    }
+
+    private initializeListeners(win: Window): void {
+        // currently assumes activity based on key/mouse/touch pressed, not on mouse move or scrolling.
+        win.addEventListener('mousedown', this.resetInactivity, eventListenerOptions);
+        win.addEventListener('keydown', this.resetInactivity, eventListenerOptions);
+        win.addEventListener('touchstart', this.resetInactivity, eventListenerOptions);
+    }
+
+    dispose(): void {
+        this.stopTracking();
+        this.win.removeEventListener('mousedown', this.resetInactivity);
+        this.win.removeEventListener('keydown', this.resetInactivity);
+        this.win.removeEventListener('touchstart', this.resetInactivity);
+
+    }
+
+    // Reset inactivity time
+    private resetInactivity = (): void => {
+        this.inactivityTime = 0;
+        if (!this.interval) {
+            // it was not active. Set as active and restart tracking inactivity
+            this.activeState = true;
+            this.startTracking();
+        }
+    };
```

Comment:
Suggestion: I'm not sure I agree with the logic here. While this resets the inactivity counter, it doesn't restart the interval. I.e. in case the user performs an action just before the interval rolls over (and thereby increases the counter by 1), the inactivity will already be registered after 10 seconds, instead of the intended 20 seconds.

I'd personally reduce the interval duration to 1 second and require 20 ticks instead of 2. This reduces the issue above to a timespan of <1s, which is effectively unnoticeable.

Reply:
adressed in new commit. Changed accordingly to your suggestion.